agents-api/agents_api/activities/task_steps/base_evaluate.py

```diff
@@ -31,10 +31,22 @@ def __init__(self, error, expression, values):
         super().__init__(message)
 
 
+# Added recursive evaluation function
+def _recursive_evaluate(expr, evaluator):
+    if isinstance(expr, str):
+        return evaluator.eval(expr)
```

Comment:
Do a try except here to pinpoint which expression caused the error. Also log the variables accessed: {name: value for name, value in vars.items() if name in expression}